Sources/SwiftDocC/Indexing/Navigator/NavigatorIndex.swift

```diff
@@ -507,10 +506,12 @@ extension NavigatorIndex {
         /// The map of the availabilities from a single availabilityID to an array of availabilities inside the availability index.
         /// This approach gives us the opportunity to map multiple availabilities with the same entries using a single ID.
         /// Ex. An item with: iOS 13.0 and macOS 10.15 can share the same ID with other items having exactly the same availability.
-        private var availabilityIDs = [Int: [Int]]()
+        /// We use the `0` value to indicate that there are no associated availabilityIndex entries.
```

Comment:
Is there a comment (just a regular non-documentation comment) we could add above these two properties (below `nameToPlatform`) to explain more generally what we these maps are used for and what the `0` value is reserved for?

Also- if it could briefly explain the bug you encountered and how the `0` fixes it.

I think I'll be confused about this next time I see it. 😃

Reply:
Not sure explaining the bug here is appropriate, but I will add a description for what these two dictionaries do exactly. Essentially what was happening is that some parts of the code assumed that an availability ID of 0 meant that there is no availability information in the `availabilityIndex` for that particular `NavigatorItem`. However, this wasn't a guarantee, because 0 wasn't reserved for that. If the first `RenderNode` to be indexed had availability information the availability data for that would be stored under ID 0. So that was why we were seeing all these symbols being marked as deprecated. Essentially all symbols that didn't define their own availability would wind up having the availability of the first RenderNode to be indexed.

Reply:
Ah got it. Agreed we don't need an explanation of the bug in that case then.

Thanks for explaining here! And I think the addition of `availabilityIDWithNoAvailabilities` makes this easy to follow.